src/monaco/browser/monaco-editor.ts

```diff
@@ -76,15 +77,38 @@ export class MonacoEditor implements TextEditor, IEditorReference {
     protected readonly onSelectionChangedEmitter = new Emitter<Range>();
     protected readonly onFocusChangedEmitter = new Emitter<boolean>();
 
+    protected readonly lineNumbers: string;
+
     constructor(
         readonly uri: URI,
         readonly node: HTMLElement,
         protected readonly m2p: MonacoToProtocolConverter,
         protected readonly p2m: ProtocolToMonacoConverter,
         protected readonly workspace: MonacoWorkspace,
+        protected readonly prefService: PreferenceService,
         options?: MonacoEditor.IOptions,
         override?: IEditorOverrideServices,
     ) {
+        this.prefService.onPreferenceChanged((pref) => {
```

Comment:
nice, if it works, but it should not appear in the final PR, just keep in your mind